### PR TITLE
ENH: add timeseries fliers

### PR DESCRIPTION
**Pull request by tacaswell**
Modified version of code @stuwilkins posted in
https://github.com/NSLS-II/wishlist/issues/77

Needs testing.

closes https://github.com/NSLS-II/wishlist/issues/77

### CODE REVIEW

File: ophyd/controls/fliers.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import epics
+from .ophydobj import SatusBase
```
Comment (klauer) [inline ophyd/controls/fliers.py]: `Satus`?

Comment (tacaswell) [inline ophyd/controls/fliers.py]: ? I don't understand the question

Comment (tacaswell) [inline ophyd/controls/fliers.py]: I figured it out, I just can't read.

**Comment by stuwilkins** [comment]
From my side ... only because I did a quick hack on a sunday to get this working.

**Comment by tacaswell** [comment]
@stuwilkins Has this exact code (which I mangled a bit) been tested on a beamline?

@dchabot can you put in a PR to my branch to switch the `epics.PV` to `Signals`?

Any one have any other comments?

**Comment by tacaswell** [comment]
closes #153

**Comment by tacaswell** [comment]
ping @klauer 

Do you want to leave these as stand-alone objects or fold them into the meta-class hierarchy ?

**Comment by danielballan** [comment]
bump @klauer

**Comment by klauer** [comment]
This should be pretty easy to move to an `ophyd.Device`. Want to make those modifications, @tacaswell?

**Comment by tacaswell** [comment]
Replaced my branch with @klauer 's

**Comment by tacaswell** [comment]
Can I review and merge this?  I think I own the white space at this point :wink: .

**Comment by klauer** [comment]
I should probably have noted that the refactor is completely untested (other than importing and device instantiation). I think @stuwilkins is the only one at this point who can verify its functionality.

**Comment by stuwilkins** [comment]
We have the simdetectorIOC running at CSX with the latest AD, should be able to test there.

**Comment by danielballan** [comment]
There are couple uses of `.put(..., wait=True)`. Is the waiting known to be reliable in this case? Should we prefer `set_and_wait`?

**Comment by danielballan** [comment]
@tacaswell The simdetectorIOC that @stuwilkins mentioned is running with the prefix `'XF:23ID1-ES{Tst-Cam:1}'`.

**Comment by klauer** [comment]
@danielballan Potentially - mainly if these PVs have corresponding `_RBV`s. A quick check seems to indicate `TSControl` and such do not. ([ref NDStats.template](https://github.com/areaDetector/ADCore/blob/b3b2ec97985f97d2fb8b30c3b2eda1ba427f45d6/ADApp/Db/NDStats.template#L295))

File: ophyd/flyers.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        status._finished()
+        return status
+
+    def collect(self):
+        self.stop()
+        payload_val, payload_time = self._get_wfrm()
+        for v, t in zip(payload_val, payload_time):
+            yield {'data': {self.name: v},
+                   'timestamps': {self.name: t},
+                   'time': ttime.time()}
```
Comment (danielballan) [inline ophyd/flyers.py]: `'time': time.time()` -> `'time': t` before this is merged

**Comment by danielballan** [comment]
bump

File: ophyd/flyers.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+class AreaDetectorTimeseriesCollector(Device):
+    ts_control = C(EpicsSignal, "TSControl")
+    ts_num_points = C(EpicsSignal, "TSNumPoints")
+    ts_cur_point = C(EpicsSignalRO, "TSCurrentPoint")
+    ts_wfrm = C(EpicsSignalRO, "TSTotal", auto_monitor=False)
+    ts_wfrm_ts = C(EpicsSignalRO, "TSTimestamp", auto_monitor=False)
+    num_points = C(Signal)
+
+    def __init__(self, prefix, *, read_attrs=None, configuration_attrs=None,
+                 monitor_attrs=None, name=None, parent=None,
```
Comment (klauer) [inline ophyd/flyers.py]: `monitor_attrs` needs to go away or this will fail now

**Comment by klauer** [comment]
Tweaking this in my repo [here](https://github.com/klauer/ophyd/commits/enh_fliers) and trying to test using 23id's sim detector - code isn't failing but not quite working either.

**Comment by klauer** [comment]
OK - AD time series collector tested and working at 23id with included test. Will wait for @tacaswell's return to merge my branch in for final review.

@stuwilkins do you have an IOC I can use to test the `WaveformCollector`?

**Comment by tacaswell** [comment]
reset my branch to the tip of @klauer s branch

**Comment by klauer** [comment]
With no input from @stuwilkins, this is as good as I can do. @tacaswell squash the commits if you could, and merge at will.

**Comment by stuwilkins** [comment]
Hey, sorry if I missed something, what would you like me to do!?!?

**Comment by klauer** [comment]
@stuwilkins no problem, I know you're busy. I made some tests for the AD time series collector that use your recent AD simulator IOC. I was hoping to write a simple `WaveformCollector` test as well - so do you have an IOC I can use to test it without interrupting anything at the beamline?

**Comment by danielballan** [comment]
bump @stuwilkins ^

**Comment by tacaswell** [comment]
@cmazzoli Can you test using these classes to see if it addresses your performance issues?

**Pull request merged**